File: 0257-二叉树的所有路径/Solution.java
import java.util.ArrayList;
import java.util.List;

class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;

    TreeNode(int x) {
        val = x;
    }
}

class Solution {
    public List<String> binaryTreePaths(TreeNode root) {
        List<String> res = new ArrayList<>();
        if (root == null) {
            return res;
        } else if (root.left == null && root.right == null) {
            res.add(String.valueOf(root.val));
            return res;
        }

        List<String> left = binaryTreePaths(root.left);
        List<String> right = binaryTreePaths(root.right);
        for (int i = 0; i < left.size(); i++) {
            res.add(root.val + "->" + left.get(i));
        }
        for (int i = 0; i < right.size(); i++) {
            res.add(root.val + "->" + right.get(i));
        }
        return res;
    }
}

File: 0257-二叉树的所有路径/solution.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def binaryTreePaths(self, root: TreeNode) -> List[str]:
        if not root:
            return []
        ret = []
        self.dfs(root, [], ret)
        ret = [ "->".join(path) for path in ret]
        return ret

    def dfs(self, root, path, ret):
        if root.left is None and root.right is None:
            path.append(str(root.val))
            ret.append(path)
            return None
        if root.left:
            self.dfs(root.left, path+[str(root.val)], ret)
        if root.right:
            self.dfs(root.right, path+[str(root.val)], ret)